src/main/c/cext/ruby.c

```diff
@@ -1464,6 +1468,11 @@ int rb_enc_str_coderange(VALUE str) {
   return truffle_invoke_i(RUBY_CEXT, "rb_enc_str_coderange", str);
 }
 
+int rb_tr_obj_equal(VALUE first, VALUE second) {
+  return rb_funcall(first, rb_intern("equal?"), 1, second);
```

Comment:
rb_funcall returns a VALUE, you need `RTEST(rb_funcall())` here.

Reply:
Thanks. This apparently worked, but I'll add the `RTEST`.

Reply:
Yes, since we unfortunately have to disable pointer/integer cast warnings we miss it easily.
But it would consider `nil` as "true", and return values > 1.